src/pages/mypage/index.tsx

```diff
@@ -75,6 +133,46 @@ const MyPage: NextPage<MyPageProps> = ({
         <div className={classes.background}>
           <div className={classes.mainContent}>
             <h2>Profil</h2>
+            <Box display="flex" justifyContent="center">
+              <Avatar
+                src={profileImage}
+                sx={{ width: 100, height: 100 }}
+                onClick={handleOpen}
+              />
+            </Box>
+            <Modal
+              disablePortal
+              open={open}
+              onClose={handleClose}
+              aria-labelledby="modal-modal-title"
+              aria-describedby="modal-modal-description"
+            >
+              <Box sx={style}>
+                <Typography id="modal-modal-title" variant="h6" component="h2">
+                  Välj profilbild
+                </Typography>
+                <ImageList
+                  sx={{ width: 500, height: 450 }}
+                  cols={3}
+                  rowHeight={164}
+                >
+                  {itemData.map((item) => (
+                    <ImageListItem key={item.img}>
+                      <img
+                        src={`${item.img}?w=164&h=164&fit=crop&auto=format`}
+                        srcSet={`${item.img}?w=164&h=164&fit=crop&auto=format&dpr=2 2x`}
+                        alt={item.title}
+                        loading="lazy"
+                        onClick={() => setProfileImage(item.img)}
```

Comment:
I tried, and it works really good but I was a bit confused if images has changed or not, so I was thinking it would be better to close the modal after you click the image.  So I think onClick, maybe write two functions would be better which is setProfileImage and handleSaveProfile.